examples/erc721/tests/erc721.rs

```diff
@@ -73,77 +134,229 @@ async fn errors_when_reusing_token_id(alice: Account) -> eyre::Result<()> {
+    let token_id = random_token_id();
+
+    let _ = watch!(contract_alice.mint(alice_addr, token_id))?;
+    let _ = watch!(contract_alice.approve(bob_addr, token_id))?;
+
+    let Erc721::balanceOfReturn { balance: initial_alice_balance } =
+        contract_alice.balanceOf(alice_addr).call().await?;
+
+    let Erc721::balanceOfReturn { balance: initial_bob_balance } =
+        contract_bob.balanceOf(bob_addr).call().await?;
+
+    let receipt =
+        receipt!(contract_bob.transferFrom(alice_addr, bob_addr, token_id))?;
+
+    assert!(receipt.emits(Erc721::Transfer {
+        from: alice_addr,
+        to: bob_addr,
+        tokenId: token_id,
+    }));
+
+    let Erc721::ownerOfReturn { ownerOf } =
+        contract_alice.ownerOf(token_id).call().await?;
+    assert_eq!(bob_addr, ownerOf);
+
+    let Erc721::balanceOfReturn { balance: alice_balance } =
+        contract_alice.balanceOf(alice_addr).call().await?;
+
+    let Erc721::balanceOfReturn { balance: bob_balance } =
+        contract_bob.balanceOf(bob_addr).call().await?;
+
+    let one = uint!(1_U256);
+    assert_eq!(initial_alice_balance - one, alice_balance);
+    assert_eq!(initial_bob_balance + one, bob_balance);
+
     Ok(())
 }
 
 #[e2e::test]
-async fn errors_when_transfer_nonexistent_token(
+async fn transfers_from_approved_for_all(
     alice: Account,
     bob: Account,
+) -> eyre::Result<()> {
+    let contract_addr = deploy(alice.url(), &alice.pk()).await?;
+    let contract_alice = Erc721::new(contract_addr, &alice.wallet);
+    let contract_bob = Erc721::new(contract_addr, &bob.wallet);
+
+    let alice_addr = alice.address();
+    let bob_addr = bob.address();
+    let token_id = random_token_id();
+
+    let _ = watch!(contract_alice.mint(alice_addr, token_id))?;
+    let _ = watch!(contract_alice.setApprovalForAll(bob_addr, true))?;
+
+    let Erc721::balanceOfReturn { balance: initial_alice_balance } =
+        contract_alice.balanceOf(alice_addr).call().await?;
+
+    let Erc721::balanceOfReturn { balance: initial_bob_balance } =
+        contract_bob.balanceOf(bob_addr).call().await?;
+
+    let receipt =
+        receipt!(contract_bob.transferFrom(alice_addr, bob_addr, token_id))?;
+
+    assert!(receipt.emits(Erc721::Transfer {
+        from: alice_addr,
+        to: bob_addr,
+        tokenId: token_id,
+    }));
+
+    let Erc721::ownerOfReturn { ownerOf } =
+        contract_alice.ownerOf(token_id).call().await?;
+    assert_eq!(bob_addr, ownerOf);
+
+    let Erc721::balanceOfReturn { balance: alice_balance } =
+        contract_alice.balanceOf(alice_addr).call().await?;
+
+    let Erc721::balanceOfReturn { balance: bob_balance } =
+        contract_bob.balanceOf(bob_addr).call().await?;
+
+    let one = uint!(1_U256);
+    assert_eq!(initial_alice_balance - one, alice_balance);
+    assert_eq!(initial_bob_balance + one, bob_balance);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn error_when_transfer_from_transfers_to_invalid_receiver(
```

Comment:
This one seems better to be renamed since this name convention doesn't work well e.g. to `error_when_transfers_to_invalid_receiver`

Reply:
And others with names like `error_when_transfer_from_transfers_from_incorrect_owner`, `error_when_transfer_from_transfers_with_insufficient_approval`, `error_when_transfer_from_transfers_from_incorrect_owner`, `error_when_safe_transfer_from_transfers_to_invalid_receiver`,
`error_when_safe_transfer_from_with_data_transfers_from_incorrect_owner`,
`error_when_safe_transfer_from_with_data_transfers_with_insufficient_approval`

Reply:
At first mine comment there is an example how it can be changed. IMHO just remove function of the method if possible and doesn't collide with other name. But ofcourse you should know better which name describes better the test case (since we are not going to use previous convention for error test cases)